### PR TITLE
Rot functions

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
Added rot, rotg, rotm, rotmg and test code for all relevant data types.

### CODE REVIEW

File: clients/include/cblas_interface.hpp [hunk 1]
```diff
@@ -365,6 +365,196 @@ inline void cblas_swap(rocblas_int             n,
...
+inline void cblas_rotmg(float* d1, float* d2, float* b1, const float* b2, float* p)
+{
+    cblas_srotmg(d1, d2, b1, *b2, p);
+}
+
+template <>
+inline void cblas_rotmg(double* d1, double* d2, double* b1, const double* b2, double* p)
+{
+    cblas_drotmg(d1, d2, b1, *b2, p);
+}
```
Comment (leekillough) [inline clients/include/cblas_interface.hpp]: This should be templatized; also, we don't provide Fortran-compatible APIS for any routines.

File: clients/include/testing_rotm.hpp [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+                << ",norm_error_host_x,norm_error_host_y,norm_error_device_x,norm_error_device_y";
+        std::cout << std::endl;
+        std::cout << N << "," << incx << "," << incy << "," << gpu_time_used << ","
+                  << cpu_time_used;
+        if(arg.norm_check)
+            std::cout << ',' << norm_error_host_x << ',' << norm_error_host_y << ","
+                      << norm_error_device_x << "," << norm_error_device_y;
+        std::cout << std::endl;
+    }
+}
```
Comment (leekillough) [inline clients/include/testing_rotm.hpp]: The testing_rot* functions should be consolidated into one file, not duplicated.

File: library/src/blas1/rocblas_rotg.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+rocblas_status rocblas_zrotg(rocblas_handle          handle,
+                             rocblas_double_complex* a,
+                             rocblas_double_complex* b,
+                             double*                 c,
+                             rocblas_double_complex* s)
+{
+    return rocblas_rotg(handle, a, b, c, s);
+}
+
```
Comment (leekillough) [inline library/src/blas1/rocblas_rotg.cpp]: The rot* functions should be consolidated into one file and differentiated by template arguments, not copied-and-edited.

Comment (AlexBrownAMD) [reply library/src/blas1/rocblas_rotg.cpp]: Not sure I follow your meaning, the rot* functions are all different operations, they are not copies of each other

File: clients/include/cblas_interface.hpp (hunk 1 above)
Comment (AlexBrownAMD) [reply clients/include/cblas_interface.hpp]: Can you provide an example? I think these are templated the same way as the rest of cblas_interface

**Comment by leekillough** [comment]
When you finally commit your changes, please use "Squash and merge" to minimize the number of commits on the main develop branch.

**Comment by AlexBrownAMD** [comment]
All tests quick+pre_checkin+nightly pass on gfx900 and gfx906

**Pull request merged**